compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -411,6 +411,18 @@ impl SirFuncCx<'tcx> {
                 self.lower_binop(bx, bb, dest_ty, *op, opnd1, opnd2, true)
             }
             mir::Rvalue::Cast(mir::CastKind::Misc, op, ty) => self.lower_cast_misc(bx, bb, op, ty),
+            mir::Rvalue::Len(p) => {
+                // Yorick's register allocator will always put a fat pointer on the stack, so
+                // finding its length is a matter of skipping to the second field of the fat
+                // pointer.
+                let ip = self.lower_place(bx, bb, p);
+                self.offset_iplace(
+                    bx,
+                    ip,
+                    i32::try_from(std::mem::size_of::<usize>()).unwrap(),
```

Comment:
This is not correct for cross-compiling. You will need to take it from `tcx.data_layout.pointer_size.bits()`.

Reply:
Thanks. Fixed in a4edfc4